[New Product] Oracle Database

adriens opened the pull request:
cf #2127

adriens committed:
First draft for oracle-database

cf #2127

marcwrobel commented on products/oracle-database.md:
```diff
@@ -0,0 +1,117 @@
+---
+title: Oracle
```
Oracle is the company name. I agree Oracle is often synonym to Oracle Database, but given endoflife.date is listing other Oracle products we should use "Oracle Database" here to not confuse the users.

marcwrobel commented on products/oracle-database.md:
```diff
@@ -0,0 +1,117 @@
+category: db
+iconSlug: oracle
+permalink: /oracle-database
+versionCommand: show version
```
I do not know in which context this command is valid. Maybe `SELECT BANNER_FULL FROM V$VERSION;` (taken from https://database.guide/7-ways-to-check-your-oracle-version/) would be better ?

marcwrobel commented on products/oracle-database.md:
```diff
@@ -0,0 +1,117 @@
+permalink: /oracle-database
+versionCommand: show version
+releasePolicyLink: https://www.oracle.com/us/assets/lifetime-support-technology-069183.pdf
+# changelogTemplate: "https://dev.mysql.com/doc/relnotes/mysql/__RELEASE_CYCLE__/en/news-{{'__LATEST__'|replace:'.','-'}}.html"
```
Release notes are complex (for example https://docs.oracle.com/cd/B19306_01/nav/portal_1.htm, https://docs.oracle.com/en/database/oracle/oracle-database/21/whats-new.html). This `changelogTemplate` comment must be removed.

marcwrobel commented on products/oracle-database.md:
```diff
@@ -0,0 +1,117 @@
+activeSupportColumn: Premier Support
+eolColumn: Extended Support
+
+# Regex takes into account the first GA release in each cycle (in parentheses)
```
Relates to mysql, should be removed.

It seems Oracle publishes docker image on https://container-registry.oracle.com/, update may be automated from this (requires a new script in https://github.com/endoflife-date/release-data/).

marcwrobel commented on products/oracle-database.md:
```diff
@@ -0,0 +1,117 @@
+releases:
+
+# https://docs.oracle.com/en/database/other-databases/essbase/21/essop/21.1.0.0.0-december-2020.html
+-   releaseCycle: "21c"
```
Link to release note should be added.

For example :
- Oracle 10g : https://docs.oracle.com/cd/B19306_01/nav/portal_1.htm
- Oracle 21c : https://docs.oracle.com/en/database/oracle/oracle-database/21/whats-new.html

marcwrobel commented on products/oracle-database.md:
```diff
@@ -0,0 +1,117 @@
+    releaseDate: 1997-06-01
+ 
+---
+
```
Missing product description.

marcwrobel replied:
Product description is the first paragraph of the description, you must not add a title.

Moreover this description must be in a single quote block.

marcwrobel commented on products/oracle-database.md:
```diff
@@ -0,0 +1,117 @@
+---
+
+
+## Oracle Lifetime Support Policy
```
Support policy should be resumed, not copy/pasted from Oracle documentation.

adriens committed:
oracle is the company name

As of https://github.com/endoflife-date/endoflife.date/pull/2476#discussion_r1104857778 

"Oracle is the company name. I agree Oracle is often synonym to Oracle Database, but given endoflife.date is listing other Oracle products we should use "Oracle Database" here to not confuse the users."

adriens committed:
Cleanup and changelog template

as of https://github.com/endoflife-date/endoflife.date/pull/2476#discussion_r1104863374

"Release notes are complex (for example https://docs.oracle.com/cd/B19306_01/nav/portal_1.htm, https://docs.oracle.com/en/database/oracle/oracle-database/21/whats-new.html). This changelogTemplate comment must be removed."

adriens committed:
Link to release note should be added.

as of https://github.com/endoflife-date/endoflife.date/pull/2476#discussion_r1104870343

"

Link to release note should be added.

For example :

    Oracle 10g : https://docs.oracle.com/cd/B19306_01/nav/portal_1.htm
    Oracle 21c : https://docs.oracle.com/en/database/oracle/oracle-database/21/whats-new.html

"

adriens committed:
Add Missing product description

as of https://github.com/endoflife-date/endoflife.date/pull/2476#discussion_r1104870871

"Missing product description."

adriens committed:
Support policy should be resumed, not copy/pasted from Oracle documentation.

As of https://github.com/endoflife-date/endoflife.date/pull/2476#discussion_r1104871439

"Support policy should be resumed, not copy/pasted from Oracle documentation."

marcwrobel commented on products/oracle-database.md:
```diff
@@ -0,0 +1,124 @@
+    eol: false
+    latestReleaseDate: 2023-01-13
+    releaseDate: 2019-02-01
+    link: "https://docs.oracle.com/en/database/oracle/oracle-database/19/rnrdm/database-release-notes.pdf"
```
HTML links, such as https://docs.oracle.com/en/database/oracle/oracle-database/19/rnrdm/index.html, should be favored IMHO.

marcwrobel commented on products/oracle-database.md:
```diff
@@ -0,0 +1,124 @@
+    eol: false
+    latestReleaseDate: 2023-01-13
+    releaseDate: 2019-02-01
+    link: "https://support.oracle.com/knowledge/Oracle%20Cloud/18202104_9.html"
```
This is the changelog for Oracle Cloud. May be better to use https://docs.oracle.com/en/database/oracle/oracle-database/18/rnrdm/index.html ?

marcwrobel commented on products/oracle-database.md:
```diff
@@ -0,0 +1,124 @@
+    eol: true
+    latestReleaseDate: 2022-09-13
+    releaseDate: 2007-09-01
+    link: "https://support.oracle.com/knowledge/Oracle%20Cloud/2068368_1.html"
```
This is the changelog for Oracle Cloud, may be better to use https://docs.oracle.com/cd/E11882_01/relnotes.112/e23558/toc.htm ?

marcwrobel commented on products/oracle-database.md:
```diff
@@ -0,0 +1,124 @@
+
+## Oracle Lifetime Support Policy
+
+> The "Lifetime Support Policy" provides access to technical experts for as long as
```
Why using quote blocks for this paragraph (and the following) ?

marcwrobel commented on products/oracle-database.md:
```diff
@@ -0,0 +1,124 @@
+
+## Premier Support
+
+> Premier Support provides you with maintenance and support for your Oracle Database
```
A more neutral and concise style should be preferred. You should avoid sentences with "you" or "your".

Example for this paragraph :

> Premier Support provides maintenance and support for five years from the general availability date of a given release.

marcwrobel commented on products/oracle-database.md:
```diff
@@ -0,0 +1,124 @@
+  
+## Extended Support
+  
+> Extended Support lets you stay competitive, with the freedom to upgrade on your
```
A more neutral style should be preferred (see comment above).

marcwrobel commented on products/oracle-database.md:
```diff
@@ -0,0 +1,124 @@
+
+## Sustaining Support
+  
+> Sustaining Support puts you in control of your upgrade strategy. When Premier Support expires, if you choose
```
A more neutral style should be preferred (see comment above).

marcwrobel commented on products/oracle-database.md:
```diff
@@ -0,0 +1,124 @@
+iconSlug: oracle
+permalink: /oracle-database
+versionCommand: show version
+releasePolicyLink: https://www.oracle.com/us/assets/lifetime-support-technology-069183.pdf
```
HTML links should be favored, such as https://www.oracle.com/support/lifetime-support/software.html.

adriens replied:
> HTML links should be favored, such as https://www.oracle.com/support/lifetime-support/software.html.

:bulb: Could this lint be part of the CI :grey_question:

marcwrobel commented on products/oracle-database.md:
```diff
@@ -0,0 +1,124 @@
+    latest: "21.4.2.0.0"
+    support: 2024-04-30
+    eol: false
+    latestReleaseDate: 2022-12-16
```
What were your source(s) for the dates ? I ask this so I can quickly validate. Do not hesitate to put, for each release, a link to the page that give the information.

Moreover, you should clarify what's being tracked here. According to https://support.oracle.com/knowledge/Oracle%20Database%20Products/742060_1.html there are _On-Premises Server Releases_ and _Oracle Public Cloud Releases_, and those releases have different release dates for the same major versions.

adriens replied:
https://github.com/endoflife-date/endoflife.date/issues/2127#issuecomment-1422069730

adriens replied:
https://github.com/endoflife-date/endoflife.date/issues/2127#issuecomment-1422070186

adriens replied:
- https://support.oracle.com/knowledge/Oracle%20Cloud/21202110_9.html
- https://support.oracle.com/knowledge/Oracle%20Cloud/2796590_1.html

adriens replied:
Also from [Wikipedia](https://en.wikipedia.org/wiki/Oracle_Database) : 

![image](https://user-images.githubusercontent.com/5235127/218881980-00c57cf0-5c01-4214-aa04-f68d051ba499.png)

adriens replied:
Also from[ Oracle Database 21c Release Update & Release Update Revision July 2022 Known Issues (Doc ID 21202207.9)](https://support.oracle.com/knowledge/Oracle%20Database%20Products/21202207_9.html)

adriens committed:
Versin command

as of https://github.com/endoflife-date/endoflife.date/pull/2476#discussion_r1104860368

I do not know in which context this command is valid. Maybe SELECT BANNER_FULL FROM V$VERSION; (taken from https://database.guide/7-ways-to-check-your-oracle-version/) would be better ?

adriens committed:
Better produit description

as of https://github.com/endoflife-date/endoflife.date/pull/2476#discussion_r1106079084



Product description is the first paragraph of the description, you must not add a title.

Moreover this description must be in a single quote block.

adriens committed:
Better link for releaseNotes

as of https://github.com/endoflife-date/endoflife.date/pull/2476#discussion_r1106069962

"HTML links, such as https://docs.oracle.com/en/database/oracle/oracle-database/19/rnrdm/index.html, should be favored IMHO."

adriens committed:
Better release link

as of https://github.com/endoflife-date/endoflife.date/pull/2476#discussion_r1106073187

"This is the changelog for Oracle Cloud, may be better to use https://docs.oracle.com/cd/E11882_01/relnotes.112/e23558/toc.htm ?"

adriens committed:
Remove quote block

as of https://github.com/endoflife-date/endoflife.date/pull/2476#discussion_r1106080617

"Why using quote blocks for this paragraph (and the following) ?"

adriens committed:
more neutral and concise style

As of https://github.com/endoflife-date/endoflife.date/pull/2476#discussion_r1106082218

"A more neutral and concise style should be preferred. You should avoid sentences with "you" or "your"."

adriens committed:
Better datas for 21c

as of https://github.com/endoflife-date/endoflife.date/pull/2476#discussion_r1106094045

adriens committed:
html release policy link

as of https://github.com/endoflife-date/endoflife.date/pull/2476#discussion_r1106086489

"HTML links should be favored, such as https://www.oracle.com/support/lifetime-support/software.html."

marcwrobel committed:
Merge product description paragraphes

marcwrobel committed:
Precise that Oracle Database is a commercial product

adriens reviewed (commented):
:pray:

marcwrobel committed:
[oracle-database] Rewrite product page

- Major description update to better explain what's documented on the page, the various support stages, the various sorts of releases...
- Remove activeSupport and latest column,
- Add extendedSupport column,
- Fix releases dates and links,
- Make Release 1 & 2 distinct.

I stopped at Oracle 9i (20 years ago). Before that versionning was a bit different, information is sparse.

marcwrobel commented on products/oracle-database.md:
```diff
@@ -0,0 +1,118 @@
+
+Oracle Database is a complex product distributed in various flavor (multiple cloud and on-premises
+offers), with support for various operating systems (such as Linux, Oracle Solaris or Microsoft
+Windows), and with [various support stages](https://www.oracle.com/support/lifetime-support/software.html "Lifetime Support for Oracle software")
```
Mention https://support.oracle.com/knowledge/Oracle%20Cloud/161818_1.html. To access this document, an Oracle account with appropriate permissions is needed.

marcwrobel reviewed (approved):
LGTM but an extensive review is required. Ping @endoflife-date/everyone ;).

captn3m0 commented on products/oracle-database.md:
```diff
@@ -0,0 +1,138 @@
+Windows), and with [various support stages](https://www.oracle.com/support/lifetime-support/software.html "Lifetime Support for Oracle software")
+(such as Premier, Extended, Sustaining or Custom Market Driven). Depending on these offers release
+dates may vary. This page is only documenting Premier and Extended support dates for Oracle Database
+On-Premises Server Releases for Linux x86-64.
```
```suggestion
dates may vary. This page documents Premier and Extended support dates for Oracle Database
On-Premises Server Releases for Linux x86-64.
```

captn3m0 commented on products/oracle-database.md:
```diff
@@ -0,0 +1,138 @@
+
+Oracle Database is a complex product distributed in various flavor (multiple cloud and on-premises
+offers), with support for various operating systems (such as Linux, Oracle Solaris or Microsoft
+Windows), and with [various support stages](https://www.oracle.com/support/lifetime-support/software.html "Lifetime Support for Oracle software")
```
```suggestion
Oracle Database has [various support stages](https://www.oracle.com/support/lifetime-support/software.html "Lifetime Support for Oracle software")
```

captn3m0 commented on products/oracle-database.md:
```diff
@@ -0,0 +1,138 @@
+- Innovation Release, which are supported for at least 2 years with Premier Support and have no
+  Extended Support.
+
+The version and patch policy is also complex and changing, but some interesting information can be
```
"interesting information" -> "guidance"

```suggestion
The version and patch policy is also complex and changing, but some guidance can be
```

marcwrobel committed:
Update products/oracle-database.md

Co-authored-by: Nemo <commits@captnemo.in>

marcwrobel committed:
Update products/oracle-database.md

Co-authored-by: Nemo <commits@captnemo.in>

marcwrobel committed:
Update products/oracle-database.md

Co-authored-by: Nemo <commits@captnemo.in>

marcwrobel committed:
[oracle-database] Use simpler releaseCycle

Based on https://www.orafaq.com/wiki/Oracle_database#Database_version_history.

captn3m0 committed:
[oracle-db] Fix duplicate releaseCycle

captn3m0 committed:
[oracle-db] Fix duplicate releaseCycle

marcwrobel committed:
Improve description

marcwrobel committed:
Reformat
